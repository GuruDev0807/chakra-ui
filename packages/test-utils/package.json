{
  "name": "@chakra-ui/test-utils",
  "version": "1.0.0-next.6",
  "private": true,
  "description": "Testing utilities for Chakra UI's components",
  "keywords": [
    "chakra",
    "test utils",
    "testing",
    "utils",
    "testing library",
    "react testing library"
  ],
  "author": "Segun Adebayo <sage@adebayosegun.com>",
  "homepage": "https://github.com/chakra-ui/chakra-ui#readme",
  "license": "MIT",
  "main": "src/index.ts",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/chakra-ui/chakra-ui.git"
  },
  "bugs": {
    "url": "https://github.com/chakra-ui/chakra-ui/issues"
  },
  "dependencies": {
    "@chakra-ui/system": "^1.0.0-next.6",
    "@chakra-ui/theme": "^1.0.0-next.6",
    "@testing-library/jest-dom": "5.1.1",
    "@testing-library/react": "10.4.5",
    "@testing-library/react-hooks": "3.2.1",
    "@testing-library/user-event": "12.0.11",
    "@types/jest-axe": "3.2.1",
    "jest-axe": "3.4.0",
    "react-test-renderer": "16.12.0"
  },
  "peerDependencies": {
    "react": "16.x"
  }
}
