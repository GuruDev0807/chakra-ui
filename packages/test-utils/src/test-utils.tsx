import { ChakraProvider } from "@chakra-ui/core"
import "@testing-library/jest-dom/extend-expect"
import {
  render as RTLRender,
  RenderOptions,
  fireEvent,
  RenderResult,
  Queries,
} from "@testing-library/react"
import * as React from "react"
import { toHaveNoViolations } from "jest-axe"
import serializer from "jest-emotion"

expect.addSnapshotSerializer(serializer)
expect.extend(toHaveNoViolations)

const AllProviders: React.FC = ({ children }) => (
  <ChakraProvider resetCSS>{children}</ChakraProvider>
)

export function render(
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "queries">,
): RenderResult

export function render(ui: React.ReactElement, options?: RenderOptions) {
  return RTLRender(ui, { wrapper: AllProviders, ...options })
}

export * from "@testing-library/react"

export {
  act as invoke,
  renderHook,
  RenderHookOptions,
  RenderHookResult,
} from "@testing-library/react-hooks"

export { default as userEvent } from "@testing-library/user-event"

export * from "jest-axe"

export const escape = (ui: HTMLElement) =>
  fireEvent.keyDown(ui, { key: "Escape", keyCode: 27 })
