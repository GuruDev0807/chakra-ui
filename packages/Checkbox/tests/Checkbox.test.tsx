import React from "react"
import { userEvent, render } from "@chakra-ui/test-utils"
import {
  Checkbox,
  CheckboxHiddenInput,
  CustomCheckbox,
} from "../src/Checkbox.base"

test("Checkbox renders correctly", () => {
  const utils = render(
    <Checkbox>
      <CheckboxHiddenInput />
      <CustomCheckbox>This is custom checkbox</CustomCheckbox>
    </Checkbox>,
  )
  expect(utils.asFragment()).toMatchSnapshot()
})

test("Uncontrolled - should check and uncheck", () => {
  const utils = render(
    <Checkbox>
      <CheckboxHiddenInput data-testid="input" />
      <CustomCheckbox>Checkbox</CustomCheckbox>
    </Checkbox>,
  )

  const input = utils.getByTestId("input")
  const checkbox = utils.getByText("Checkbox")

  // click the first time, it's checked
  userEvent.click(checkbox)
  expect(input).toBeChecked()
  expect(checkbox).toHaveAttribute("data-checked")

  // click the second time, it's unchecked
  userEvent.click(checkbox)
  expect(input).not.toBeChecked()
  expect(checkbox).not.toHaveAttribute("data-checked")
})

test("Uncontrolled - should not check if disabled", () => {
  const utils = render(
    <Checkbox isDisabled>
      <CheckboxHiddenInput data-testid="input" />
      <CustomCheckbox>Checkbox</CustomCheckbox>
    </Checkbox>,
  )

  const input = utils.getByTestId("input")
  const checkbox = utils.getByText("Checkbox")

  expect(input).toBeDisabled()
  expect(checkbox).toHaveAttribute("data-disabled")

  userEvent.click(checkbox)

  expect(input).not.toBeChecked()
  expect(checkbox).not.toHaveAttribute("data-checked")
})
