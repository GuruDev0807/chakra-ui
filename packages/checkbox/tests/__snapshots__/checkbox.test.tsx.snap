// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox renders correctly 1`] = `
<DocumentFragment>
  .emotion-1 {
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  vertical-align: top;
  position: relative;
}

.emotion-0 {
  -webkit-transition: box-shadow 250ms;
  transition: box-shadow 250ms;
  border: 2px solid;
  border-radius: 0.125rem;
  border-color: inherit;
  color: #fff;
  width: 1rem;
  height: 1rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  vertical-align: top;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: -webkit-transform 240ms,opacity 240ms;
  -webkit-transition: transform 240ms,opacity 240ms;
  transition: transform 240ms,opacity 240ms;
}

.emotion-0[aria-checked=true],
.emotion-0[data-checked],
.emotion-0[data-state=checked] {
  background: #3182ce;
  border-color: #3182ce;
  color: #fff;
}

.emotion-0[aria-checked=true]:hover,
.emotion-0[data-checked]:hover,
.emotion-0[data-state=checked]:hover,
.emotion-0[aria-checked=true][data-hover],
.emotion-0[data-checked][data-hover],
.emotion-0[data-state=checked][data-hover],
.emotion-0[aria-checked=true][data-state=hover],
.emotion-0[data-checked][data-state=hover],
.emotion-0[data-state=checked][data-state=hover] {
  background: #2a69ac;
  border-color: #2a69ac;
}

.emotion-0[aria-checked=true][disabled],
.emotion-0[data-checked][disabled],
.emotion-0[data-state=checked][disabled],
.emotion-0[aria-checked=true][disabled]:focus,
.emotion-0[data-checked][disabled]:focus,
.emotion-0[data-state=checked][disabled]:focus,
.emotion-0[aria-checked=true][disabled]:hover,
.emotion-0[data-checked][disabled]:hover,
.emotion-0[data-state=checked][disabled]:hover,
.emotion-0[aria-checked=true][aria-disabled=true],
.emotion-0[data-checked][aria-disabled=true],
.emotion-0[data-state=checked][aria-disabled=true],
.emotion-0[aria-checked=true][aria-disabled=true]:focus,
.emotion-0[data-checked][aria-disabled=true]:focus,
.emotion-0[data-state=checked][aria-disabled=true]:focus,
.emotion-0[aria-checked=true][aria-disabled=true]:hover,
.emotion-0[data-checked][aria-disabled=true]:hover,
.emotion-0[data-state=checked][aria-disabled=true]:hover,
.emotion-0[aria-checked=true][data-disabled],
.emotion-0[data-checked][data-disabled],
.emotion-0[data-state=checked][data-disabled],
.emotion-0[aria-checked=true][data-disabled]:focus,
.emotion-0[data-checked][data-disabled]:focus,
.emotion-0[data-state=checked][data-disabled]:focus,
.emotion-0[aria-checked=true][data-disabled]:hover,
.emotion-0[data-checked][data-disabled]:hover,
.emotion-0[data-state=checked][data-disabled]:hover,
.emotion-0[aria-checked=true][data-state=disabled],
.emotion-0[data-checked][data-state=disabled],
.emotion-0[data-state=checked][data-state=disabled],
.emotion-0[aria-checked=true][data-state=disabled]:focus,
.emotion-0[data-checked][data-state=disabled]:focus,
.emotion-0[data-state=checked][data-state=disabled]:focus,
.emotion-0[aria-checked=true][data-state=disabled]:hover,
.emotion-0[data-checked][data-state=disabled]:hover,
.emotion-0[data-state=checked][data-state=disabled]:hover {
  border-color: #E2E8F0;
  background: #E2E8F0;
  color: #718096;
}

.emotion-0:indeterminate,
.emotion-0[aria-checked=mixed],
.emotion-0[data-indeterminate],
.emotion-0[data-state=mixed] {
  background: #3182ce;
  border-color: #3182ce;
  color: #fff;
}

.emotion-0[disabled],
.emotion-0[disabled]:focus,
.emotion-0[disabled]:hover,
.emotion-0[aria-disabled=true],
.emotion-0[aria-disabled=true]:focus,
.emotion-0[aria-disabled=true]:hover,
.emotion-0[data-disabled],
.emotion-0[data-disabled]:focus,
.emotion-0[data-disabled]:hover,
.emotion-0[data-state=disabled],
.emotion-0[data-state=disabled]:focus,
.emotion-0[data-state=disabled]:hover {
  background: #EDF2F7;
  border-color: #EDF2F7;
}

.emotion-0:focus,
.emotion-0[data-focus],
.emotion-0[data-state=focused] {
  box-shadow: 0 0 0 3px rgba(66,153,225,0.6);
}

.emotion-0[aria-invalid=true],
.emotion-0[data-invalid],
.emotion-0[data-state=invalid] {
  border-color: #e53e3e;
}

<label
    class="chakra-checkbox emotion-1"
  >
    <input
      class="chakra-checkbox__input"
      style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
      type="checkbox"
      value=""
    />
    <div
      aria-hidden="true"
      class="chakra-checkbox__control emotion-0"
    />
  </label>
</DocumentFragment>
`;

exports[`Checkbox renders correctly 2`] = `
<DocumentFragment>
  .emotion-2 {
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  vertical-align: top;
  position: relative;
}

.emotion-0 {
  -webkit-transition: box-shadow 250ms;
  transition: box-shadow 250ms;
  border: 2px solid;
  border-radius: 0.125rem;
  border-color: inherit;
  color: #fff;
  width: 1rem;
  height: 1rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  vertical-align: top;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: -webkit-transform 240ms,opacity 240ms;
  -webkit-transition: transform 240ms,opacity 240ms;
  transition: transform 240ms,opacity 240ms;
}

.emotion-0[aria-checked=true],
.emotion-0[data-checked],
.emotion-0[data-state=checked] {
  background: #3182ce;
  border-color: #3182ce;
  color: #fff;
}

.emotion-0[aria-checked=true]:hover,
.emotion-0[data-checked]:hover,
.emotion-0[data-state=checked]:hover,
.emotion-0[aria-checked=true][data-hover],
.emotion-0[data-checked][data-hover],
.emotion-0[data-state=checked][data-hover],
.emotion-0[aria-checked=true][data-state=hover],
.emotion-0[data-checked][data-state=hover],
.emotion-0[data-state=checked][data-state=hover] {
  background: #2a69ac;
  border-color: #2a69ac;
}

.emotion-0[aria-checked=true][disabled],
.emotion-0[data-checked][disabled],
.emotion-0[data-state=checked][disabled],
.emotion-0[aria-checked=true][disabled]:focus,
.emotion-0[data-checked][disabled]:focus,
.emotion-0[data-state=checked][disabled]:focus,
.emotion-0[aria-checked=true][disabled]:hover,
.emotion-0[data-checked][disabled]:hover,
.emotion-0[data-state=checked][disabled]:hover,
.emotion-0[aria-checked=true][aria-disabled=true],
.emotion-0[data-checked][aria-disabled=true],
.emotion-0[data-state=checked][aria-disabled=true],
.emotion-0[aria-checked=true][aria-disabled=true]:focus,
.emotion-0[data-checked][aria-disabled=true]:focus,
.emotion-0[data-state=checked][aria-disabled=true]:focus,
.emotion-0[aria-checked=true][aria-disabled=true]:hover,
.emotion-0[data-checked][aria-disabled=true]:hover,
.emotion-0[data-state=checked][aria-disabled=true]:hover,
.emotion-0[aria-checked=true][data-disabled],
.emotion-0[data-checked][data-disabled],
.emotion-0[data-state=checked][data-disabled],
.emotion-0[aria-checked=true][data-disabled]:focus,
.emotion-0[data-checked][data-disabled]:focus,
.emotion-0[data-state=checked][data-disabled]:focus,
.emotion-0[aria-checked=true][data-disabled]:hover,
.emotion-0[data-checked][data-disabled]:hover,
.emotion-0[data-state=checked][data-disabled]:hover,
.emotion-0[aria-checked=true][data-state=disabled],
.emotion-0[data-checked][data-state=disabled],
.emotion-0[data-state=checked][data-state=disabled],
.emotion-0[aria-checked=true][data-state=disabled]:focus,
.emotion-0[data-checked][data-state=disabled]:focus,
.emotion-0[data-state=checked][data-state=disabled]:focus,
.emotion-0[aria-checked=true][data-state=disabled]:hover,
.emotion-0[data-checked][data-state=disabled]:hover,
.emotion-0[data-state=checked][data-state=disabled]:hover {
  border-color: #E2E8F0;
  background: #E2E8F0;
  color: #718096;
}

.emotion-0:indeterminate,
.emotion-0[aria-checked=mixed],
.emotion-0[data-indeterminate],
.emotion-0[data-state=mixed] {
  background: #3182ce;
  border-color: #3182ce;
  color: #fff;
}

.emotion-0[disabled],
.emotion-0[disabled]:focus,
.emotion-0[disabled]:hover,
.emotion-0[aria-disabled=true],
.emotion-0[aria-disabled=true]:focus,
.emotion-0[aria-disabled=true]:hover,
.emotion-0[data-disabled],
.emotion-0[data-disabled]:focus,
.emotion-0[data-disabled]:hover,
.emotion-0[data-state=disabled],
.emotion-0[data-state=disabled]:focus,
.emotion-0[data-state=disabled]:hover {
  background: #EDF2F7;
  border-color: #EDF2F7;
}

.emotion-0:focus,
.emotion-0[data-focus],
.emotion-0[data-state=focused] {
  box-shadow: 0 0 0 3px rgba(66,153,225,0.6);
}

.emotion-0[aria-invalid=true],
.emotion-0[data-invalid],
.emotion-0[data-state=invalid] {
  border-color: #e53e3e;
}

.emotion-1 {
  font-size: 1rem;
  margin-left: 0.5rem;
}

.emotion-1[disabled],
.emotion-1[disabled]:focus,
.emotion-1[disabled]:hover,
.emotion-1[aria-disabled=true],
.emotion-1[aria-disabled=true]:focus,
.emotion-1[aria-disabled=true]:hover,
.emotion-1[data-disabled],
.emotion-1[data-disabled]:focus,
.emotion-1[data-disabled]:hover,
.emotion-1[data-state=disabled],
.emotion-1[data-state=disabled]:focus,
.emotion-1[data-state=disabled]:hover {
  opacity: 0.4;
}

<label
    class="chakra-checkbox emotion-2"
  >
    <input
      class="chakra-checkbox__input"
      style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
      type="checkbox"
      value=""
    />
    <div
      aria-hidden="true"
      class="chakra-checkbox__control emotion-0"
    />
    <div
      class="chakra-checkbox__label emotion-1"
    >
      This is custom checkbox
    </div>
  </label>
</DocumentFragment>
`;
